tests/integration/test_task.py

```diff
@@ -80,3 +83,229 @@ def _run(self, input_files):
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        _run_tasks_sequential(tasks)
+    exp_mssg = "diag_cow.py failed with return code 1"
+    assert exp_mssg in str(err_mssg.value)
+
+
+def test_py_diagnostic_run_parallel_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask parallel with bad Python script."""
+    diagnostic_text = "import os\n\nprint(cow)"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        for task in tasks:
+            task.run()
+    exp_mssg = "diag_cow.py failed with return code 1"
+    assert exp_mssg in str(err_mssg.value)
+
+
+def test_py_diagnostic_run_parallel_task(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK Python script."""
+    diagnostic_text = "import os\n\nprint('cow')"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    for task in tasks:
+        task.run()
+
+
+def test_ncl_diagnostic_run_parallel_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK NCL script."""
+    diagnostic_text = "cows on the [river]"
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    if shutil.which('ncl') is not None:
+        tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'ncl')
+
+        monkeypatch.setattr(BaseTask, '_run', _run)
+        with pytest.raises(DiagnosticError) as err_mssg:
+            for task in tasks:
+                task.run()
+        exp_mssg_1 = "An error occurred during execution of NCL script"
+        exp_mssg_2 = "diag_cow.ncl"
+        assert exp_mssg_1 in str(err_mssg.value)
+        assert exp_mssg_2 in str(err_mssg.value)
+
+
+def test_ncl_diagnostic_run_parallel_task(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK NCL script."""
+    diagnostic_text = _py2ncl({'cow': 22}, 'tas')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    if shutil.which('ncl') is not None:
+        tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'ncl')
+        monkeypatch.setattr(BaseTask, '_run', _run)
+        for task in tasks:
+            task.run()
+
+
+def test_r_diagnostic_run_parallel_task(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK NCL script."""
+    diagnostic_text = 'var0 <- "zg"'
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    if shutil.which('Rscript') is not None:
+        tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'R')
+        monkeypatch.setattr(BaseTask, '_run', _run)
+        for task in tasks:
+            task.run()
```

Comment:
It would be much shorter to use `pytest.mark.parametrize` to combine these two tests (and `test_py_diagnostic_run_parallel_task` above) into a single parametrized test.